Sources/SwiftParser/Attributes.swift

```diff
@@ -365,7 +365,7 @@ extension Parser {
       }
     case nil:
       return parseAttribute(argumentMode: .customAttribute) { parser in
-        let arguments = parser.parseArgumentListElements(pattern: .none)
+        let arguments = parser.parseArgumentListElements(pattern: .none, allowTrailingComma: false)
```

Comment:
So I'm revisiting **argumentListElements** usages and those are all that I've tracked: subscripts, attributes, macro roles, macro expansions and string literals.

Do you see any problems regardless of trailing comma in **string literals**?

Besides these there's one more that I can't figure out in [parseKeyPathExpression](https://github.com/apple/swift-syntax/blob/ec4e75afe680ac43540ce5412ea8bc294543c513/Sources/SwiftParser/Expressions.swift#L1013).

Reply:
I don’t see an inherent problem with allowing trailing commas in string interpolation. Could we maybe keep it as a separate, follow-up PR though? We’ve iterated on this one quite a bit already and I think it would be easier if we could focus on landing this one first and then consider the implications of expanding trailing closures to other areas in a separate PR.

And the following should be an example of a key path subscript that contains a comma:

```swift
struct Foo {
  subscript(x: Int, y y: Int) -> Int { 1 }
}

func test() {
  let _: KeyPath<Foo, Int> = \.[1, y: 2]
}
```

Reply:
So we leave the PR as it is?

Reply:
Yes, that would be my suggestion.